doc/source/how-to-run-simulations.rst

```diff
@@ -1,291 +1,338 @@
-    # each client gets 2xCPUs and half a GPU. (with a single GPU, 2 clients run concurrently)
-    my_client_resources = {"num_cpus": 2, "num_gpus": 0.5}
-    # 10 client can run concurrently on a single GPU, but only if you have 20 CPU threads.
-    my_client_resources = {"num_cpus": 2, "num_gpus": 0.1}
+    [tool.flwr.federations.local-simulation]
+    options.num-supernodes = 10
+    options.backend.client-resources.num-cpus = 1 # each ClientApp assumes to use 1 CPU (default is 2)
+    options.backend.client-resources.num-gpus = 0.0 # no GPU access to the ClientApp (default is 0.0)
 
-    # Launch the simulation
-    hist = fl.simulation.start_simulation(
-        # ...
-        client_resources=my_client_resources  # A Python dict specifying CPU/GPU resources
-    )
+With the above backend settings, your simulation will run as many ``ClientApps`` in
+parallel as CPUs you have in your system. GPU resources for your ``ClientApp`` can be
+assigned by specifying the **ratio** of VRAM each should make use of.
 
-While the ``client_resources`` can be used to control the degree of concurrency in your
-FL simulation, this does not stop you from running dozens, hundreds or even thousands of
-clients in the same round and having orders of magnitude more `dormant` (i.e. not
-participating in a round) clients. Let's say you want to have 100 clients per round but
-your system can only accommodate 8 clients concurrently. The ``VirtualClientEngine``
-will schedule 100 jobs to run (each simulating a client sampled by the strategy) and
-then will execute them in a resource-aware manner in batches of 8.
+.. code-block:: toml
 
-To understand all the intricate details on how resources are used to schedule FL clients
-and how to define custom resources, please take a look at the `Ray documentation
-<https://docs.ray.io/en/latest/ray-core/scheduling/resources.html>`_.
+    [tool.flwr.federations.local-simulation]
+    options.num-supernodes = 10
+    options.backend.client-resources.num-cpus = 1 # each ClientApp assumes to use 1 CPU (default is 2)
+    options.backend.client-resources.num-gpus = 0.25 # each ClientApp uses 25% of VRAM (default is 0.0)
+
+.. note::
+
+    If you are using TensorFlow, you need to `enable memory growth
+    <https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth>`_ so multiple
+    ``ClientApp`` instances can share a GPU. This needs to be done before launching the
+    simulation. To do so, set the environment variable
+    ``TF_FORCE_GPU_ALLOW_GROWTH="1"``.
+
+Let's see how the above configuration results in a different number of ``ClientApps``
+running in parallel depending on the resources available in your system. If your system
+has:
+
+- 10x CPUs and 1x GPU: at most 4 ``ClientApps`` will run in parallel since each requires
+  25% of the available VRAM.
+- 10x CPUs and 2x GPUs: at most 8 ``ClientApps`` will run in parallel.
+- 6x CPUs and 2x GPUs: at most 6 ``ClientApps`` will run in parallel.
```

Comment:
```suggestion
- 6x CPUs and 4x GPUs: at most 6 ``ClientApps`` will run in parallel (CPU-limited).
```

Should we increase this to `4x GPU` to contrast with the previous VRAM-limited example?